modules/services/dunst.nix

```diff
@@ -85,75 +89,87 @@ in
   };
 
   config = mkIf cfg.enable (
-    mkMerge [
-      {
-        xdg.dataFile."dbus-1/services/org.knopwob.dunst.service".source =
-          "${pkgs.dunst}/share/dbus-1/services/org.knopwob.dunst.service";
-
-        services.dunst.settings.global.icon_folders =
-          let
-            useCustomTheme =
-              cfg.iconTheme.package != hicolorTheme.package
-              || cfg.iconTheme.name != hicolorTheme.name
-              || cfg.iconTheme.size != hicolorTheme.size;
-
-            basePaths = [
-              "/run/current-system/sw"
-              "${config.home.homeDirectory}/.nix-profile"
-              cfg.iconTheme.package
-            ] ++ optional useCustomTheme hicolorTheme.package;
-
-            themes =
-              [
-                cfg.iconTheme
-              ] ++ optional useCustomTheme (
-                hicolorTheme // { size = cfg.iconTheme.size; }
-              );
-
-            categories = [
-              "actions"
-              "animations"
-              "apps"
-              "categories"
-              "devices"
-              "emblems"
-              "emotes"
-              "filesystem"
-              "intl"
-              "mimetypes"
-              "places"
-              "status"
-              "stock"
-            ];
-          in
-            concatStringsSep ":" (
-              concatMap (theme:
-                concatMap (basePath:
-                  map (category:
-                    "${basePath}/share/icons/${theme.name}/${theme.size}/${category}"
-                  ) categories
-                ) basePaths
-              ) themes
+    {
+      xdg.configFile."dunst/dunstrc".source = configFile;
```

Comment:
This will unconditionally write the `dunstrc` file, which we don't want if the user does not set any settings since they then probably have they own `dunstrc`.